### DISCUSSION TITLE
`BPF_F_KPROBE_MULTI_RETURN` was deleted on `golang.org/x/sys v0.14.0`

### DISCUSSION THREAD

**Discussion by keisku** [cilium/ebpf#1205] [Q&A] [answered]
I am asking this question out of personal curiosity. There are no negative effects occurring at the moment.

https://github.com/golang/sys/commit/1168e2596da3c107050a45359d2a33b6ff4b5bc6 deleted `BPF_F_KPROBE_MULTI_RETURN` in `golang/sys/unix/zerrors_linux.go` to fix https://github.com/golang/go/issues/63833

Therefore, https://github.com/cilium/ebpf/blob/v0.12.2/internal/unix/types_linux.go#L41 causes `../../../go/pkg/mod/github.com/cilium/ebpf@v0.12.2/internal/unix/types_linux.go:41:36: undefined: linux.BPF_F_KPROBE_MULTI_RETURN` when using `bpf2go` if we update `golang.org/x/sys v0.6.0` to `v0.14.0`.

Of course, we can use `bpf2go` without any problems if I switch to `golang.org/x/sys v0.6.0` in `go.mod`.

I am wondering if there is any negative impacts when I use Linux v6.6+ and `cilium/ebpf`.
FYI, we can still see `BPF_F_KPROBE_MULTI_RETURN` in https://github.com/torvalds/linux/blob/v6.6/include/uapi/linux/bpf.h#L1188-L1193

**Comment by mythi** [comment]
> [golang/sys@1168e25](https://github.com/golang/sys/commit/1168e2596da3c107050a45359d2a33b6ff4b5bc6) deleted `BPF_F_KPROBE_MULTI_RETURN` in `golang/sys/unix/zerrors_linux.go` to fix [golang/go#63833](https://github.com/golang/go/issues/63833)

This will trigger errors where both `cilium/ebpf` and `golang.org/x/sys v0.14.0` are used, AFAICS. I'm already seeing dependabot updates that trigger

```
 # github.com/cilium/ebpf/internal/unix
Error: vendor/github.com/cilium/ebpf/internal/unix/types_linux.go:41:36: undefined: unix.BPF_F_KPROBE_MULTI_RETURN
```

I wonder if the removal happened by accident?

**Reply by brycekahle** [reply, marked as answer]
It must be a side-effect of this change https://github.com/torvalds/linux/commit/c5487f8d91868eeab17a59cf4d164ea113f90252 which changes from a `#define` to an `enum`.

**Comment by brycekahle** [comment]
I filed an issue https://github.com/golang/go/issues/63969

**Comment by ti-mo** [comment]
Please upgrade to v0.12.3 to work around this issue.
